File: tools/hello_node/hello.js
// Hello-world test tool for the Node.js runtime.
// Spec §6 Phase 6 checkpoint: test arg-with-spaces passing.

function parseArgs(argv) {
    const params = {};
    for (let i = 2; i < argv.length; i += 2) {
        const key = argv[i].replace(/^--/, '');
        params[key] = argv[i + 1] || '';
    }
    return params;
}

const args = parseArgs(process.argv);

if (!args.path) {
    console.error('[ERROR] No --path specified');
    process.exit(1);
}

console.log('PROGRESS:0');
console.log(`[OK] Hello from Node ${process.version}`);
console.log(`[OK] path_arg:   ${args.path}`);
console.log('PROGRESS:50');

if (args.path.includes(' ')) {
    console.log('[OK] Path contains spaces - preserved correctly.');
} else {
    console.log('[WARN] Path has no spaces - try a path with spaces to really test.');
}

console.log('PROGRESS:100');
console.log('[OK] Done.');

File: tools/hello_python/hello.py
"""Hello-world test tool for the Python runtime.

Spec §6 Phase 6 checkpoint: "test argument passing with a path containing
spaces (``C:\\Test Folder (2026)\\``) for each language."

Prints the protocol tokens (PROGRESS:N, [OK], etc.) so the host UI can
exercise its full parsing path end-to-end.
"""
import argparse
import sys


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--path", required=True)
    args = parser.parse_args()

    print("PROGRESS:0", flush=True)
    print(f"[OK] Hello from Python {sys.version.split()[0]}", flush=True)
    print(f"[OK] python_exe: {sys.executable}", flush=True)
    print(f"[OK] path_arg:   {args.path}", flush=True)
    print("PROGRESS:50", flush=True)

    if " " in args.path:
        print(f"[OK] Path contains spaces — preserved correctly.", flush=True)
    else:
        print(f"[WARN] Path has no spaces — try a path with spaces to really test.", flush=True)

    print("PROGRESS:100", flush=True)
    print("[OK] Done.", flush=True)


if __name__ == "__main__":
    main()
